### ISSUE TITLE
Transparency Regarding Cluster State

### ISSUE THREAD

**Issue by cornelius-keller** [giantswarm/roadmap#75]
## User story

These are exmples:

- As a cluster-admin I like to see how many nodes of my new cluster are already available.

- As a cluster-admin I like to know how many nodes are actually running within a cluster, what dimensions (RAM, CPU) and what versions they have.

## Rationale

We need to distinguish between desired and actual state. This is especially important for:

* cluster creation
* cluster scaling
* upgrades
* cluster deletion

## Background

At the moment our UIs reflect a desired state for a cluster. For a better understhttps://github.com/giantswarm/roadmap/issues/57anding of the cluster we would also need to show our users the current state. Knowing the state would also enable us to add some more features.

## Open questions

How deep do we wanna go regarding the state of the cluster? Where do we draw the line between actual state within our API/UIs as opposed to prometheus monitoring and Grafana.

## Related stories

History of a Cluster: https://github.com/giantswarm/roadmap/issues/57
Real Time Event Stream: https://github.com/giantswarm/roadmap/issues/56

**Comment by teemow** [comment]
@puja108 @marians also very similar to #13 and relevant for fleet management

**Comment by marians** [comment]
I'm putting these three related issues on Honeybadger's board for Backstage.

- https://github.com/giantswarm/roadmap/issues/13
- https://github.com/giantswarm/roadmap/issues/75
- https://github.com/giantswarm/roadmap/issues/57

**Comment by marians** [comment]
Regarding this issue:

- We display cluster state in our Backstage UI (e. g. https://devportal.giantswarm.io/clusters) and both `Ready` and `Deleting` definitely appear there already
- Not sure whether we also show anything regarding upgrading or creating.
- I don't think we should show anything regarding scaling in that context, as it's just a normal thing for a K8s cluster, separate from Readiness

**Comment by puja108** [comment]
I agree that scaling is not needed. I'd say we should check if other CAPI states like creating or upgrading (is that even a CAPI state?) also show up and if we have that we can close this (we should also document where people can see this, also in the CRs, but that might rather be a topic for @giantswarm/team-turtles).
